griptape/drivers/embedding/lm_studio_embedding_driver.py

```diff
@@ -0,0 +1,17 @@
+from __future__ import annotations
+from attrs import define, field, Factory
+import openai
+
+from griptape.drivers import OpenAiEmbeddingDriver
+
+
+@define
+class LmStudioEmbeddingDriver(OpenAiEmbeddingDriver):
+    """
+    Attributes:
+        base_url: API URL. Defaults to LM Studio's v1 API URL.
+        client: Optionally provide custom `openai.OpenAI` client.
+    """
+
+    base_url: str = field(default="http://localhost:1234/v1", kw_only=True, metadata={"serializable": True})
```

Comment:
I've been getting feedback from users that localhost doesn't always resolve for them (hit this with lmstudio and ollama). I had to change my default to 127.0.0.1 in order to resolve it.

Reply:
Interesting! Even the LM Studio code snippets use localhost. Updated though.
```bash
curl http://localhost:1234/v1/chat/completions \
  -H "Content-Type: application/json" \
  -d '{ 
    "model": "lmstudio-community/Meta-Llama-3-8B-Instruct-GGUF",
    "messages": [ 
      { "role": "system", "content": "Always answer in rhymes." },
      { "role": "user", "content": "Introduce yourself." }
    ], 
    "temperature": 0.7, 
    "max_tokens": -1,
    "stream": true
}'
```